### PR TITLE
Add support for secret files

### PR DESCRIPTION
**Pull request by mcardosos**

### CODE REVIEW

**Comment by mcardosos** [comment]
Example: https://github.com/Azure-Samples/azure-sdk-for-go-samples/pull/176

File: models/droidMetadata.go [hunk 1]
```diff
@@ -9,11 +9,17 @@ import (
...
+	Version      string                 `yaml:"version"`
+	Product      string                 `yaml:"product"`
+	Storage      bool                   `yaml:"storage"`
+	Environments []DroidMetadataEnvDef  `yaml:"environments"`
+	SecretFiles  []DroidMetadataFileDef `yaml:"secretFiles"`
+}
+
+type DroidMetadataFileDef struct {
+	Path  string `yaml:"path"`
+	Value string `yaml:"value"`
```
Comment (troydai) [inline models/droidMetadata.go]: `SecretKey`

File: models/droidMetadata.go [hunk 2]
```diff
@@ -9,11 +9,17 @@ import (
...
-	Environments []DroidMetadataEnvDef `yaml:"environments"`
+	Kind         string                 `yaml:"kind"`
+	Version      string                 `yaml:"version"`
+	Product      string                 `yaml:"product"`
+	Storage      bool                   `yaml:"storage"`
+	Environments []DroidMetadataEnvDef  `yaml:"environments"`
+	SecretFiles  []DroidMetadataFileDef `yaml:"secretFiles"`
+}
+
+type DroidMetadataFileDef struct {
```
Comment (troydai) [inline models/droidMetadata.go]: Add comments about this

File: dispatcher/main.go [hunk 3]
```diff
@@ -201,6 +201,27 @@ func getVolumes(run *models.Run) (volumes []corev1.Volume) {
 			},
 		})
 
+	if len(droidMetadata.SecretFiles) == 0 {
+		return
+	}
```
Comment (troydai) [inline dispatcher/main.go]: It should be unnecessary to check the size of the secret files. If the collection is empty, the following loop will do nothing.

File: dispatcher/main.go [hunk 4]
```diff
@@ -201,6 +201,27 @@ func getVolumes(run *models.Run) (volumes []corev1.Volume) {
...
+							{
+								Key:  file.Value,
+								Path: file.Path,
+							},
+						},
+					},
+				},
+			})
+	}
 	return
```
Comment (troydai) [inline dispatcher/main.go:221]: Do we need this?

**Review by troydai — CHANGES_REQUESTED** [review]

File: dispatcher/main.go (hunk 4 above)
Comment (mcardosos) [reply dispatcher/main.go:221]: Yes, documentation [here](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod)

File: dispatcher/main.go (hunk 3 above)
Comment (mcardosos) [reply dispatcher/main.go]: Fixed

File: models/droidMetadata.go (hunk 2 above)
Comment (mcardosos) [reply models/droidMetadata.go]: Fixed

File: models/droidMetadata.go (hunk 1 above)
Comment (mcardosos) [reply models/droidMetadata.go]: Fixed

File: dispatcher/main.go (hunk 4 above)
Comment (troydai) [reply dispatcher/main.go:221]: no. I meant do we need the additional `return` since the function comes to a natural end here.

**Review by troydai — APPROVED** [review]

File: dispatcher/main.go (hunk 4 above)
Comment (mcardosos) [reply dispatcher/main.go:221]: We need it, yes

Comment (troydai) [reply dispatcher/main.go:221]: OK!

**Pull request merged**